torch/distributed/distributed_c10d.py

```diff
@@ -753,10 +755,35 @@ def recv(tensor,
         return src
 
 
+def async_collective(func):
+    @functools.wraps(func)
+    def wrapped(*args, async_op=False, **kwargs):
+        work = func(*args, **kwargs)
+        if work is None:
```

Comment:
What are the semantics here? If I call `work.wait()` in a collective and decorate with `@async_collective`, will it not be profiled correctly? If we want to use `async_collective` would the function have to return a `work` object?